src/shim.c

```diff
@@ -954,6 +958,52 @@ connect_to_proxy(struct cc_shim *shim)
 	return false;
 }
 
+inline void sleep_ms(int ms)
+{
+	struct timespec ts = { 0, ms * 1000000L };
+	nanosleep(&ts, NULL);
+}
+
+/*
+ * Try to re-establish tcp/unix connection with proxy in RECONNECT_TIMEOUT_S
+ * seconds.
+ *
+ * \param shim \ref cc_shim
+ *
+ * \return true on success, false on failure
+ */
+bool reconnect_to_proxy(struct cc_shim *shim)
+{
+	// How many seconds shim tries to reconnect to proxy
+	const int RECONNECT_TIMEOUT_S = 10;
+	// Period between each attempt in milliseconds
+	const int RECONNECT_POLL_MS = 333;
```

Comment:
It might be useful if these two variables could be specified as getopt options, defaulting to `#define`'d values you're using here (as that would allow the defaults to be modified at build time and runtime).